### DISCUSSION TITLE
Updating Open-WebUI with Ollama bundled support.

### DISCUSSION THREAD

**Discussion by yuchenxie4645** [open-webui/open-webui#2530] [General]
I checked Open-WebUI updating guide, however, it looks like they only have guides that updates the image of Open-WebUI. Normally, this would be fine, however, I could not get image only version of Open-WebUI to work, therefore I rely on Ollama's bundled support. When I followed the steps, I could not get model pulling to work, therefore, is there anyway I can update Open-WebUI with ollama bundled support?

**Comment by alexmcgilvery** [comment]
Run `docker run --rm --volume /var/run/docker.sock:/var/run/docker.sock containrrr/watchtower --run-once ollama` to update ollama.
Rub `docker run --rm --volume /var/run/docker.sock:/var/run/docker.sock containrrr/watchtower --run-once open-webui` to update the webui.

**Reply by Delivator** [reply]
> Run `docker run --rm --volume /var/run/docker.sock:/var/run/docker.sock containrrr/watchtower --run-once ollama` to update ollama. Rub `docker run --rm --volume /var/run/docker.sock:/var/run/docker.sock containrrr/watchtower --run-once open-webui` to update the webui.

Does not work for me. When installing open-webui with ollama bundled, it only creates one container called `open-webui` so using the watchtower on `ollama` does nothing. And checking in the WebUI the bundeled ollama version currently is `v0.3.14` but I need a newer version for models with vision support!

**Reply by yuchenxie4645** [reply]
> > Run `docker run --rm --volume /var/run/docker.sock:/var/run/docker.sock containrrr/watchtower --run-once ollama` to update ollama. Rub `docker run --rm --volume /var/run/docker.sock:/var/run/docker.sock containrrr/watchtower --run-once open-webui` to update the webui.
> 
> Does not work for me. When installing open-webui with ollama bundled, it only creates one container called `open-webui` so using the watchtower on `ollama` does nothing. And checking in the WebUI the bundeled ollama version currently is `v0.3.14` but I need a newer version for models with vision support!

With ollama bundled support, ollama is updated along with open webui. Open WebUI currently does not support v0.4.0  with the bundled support yet.

**Comment by Delivator** [comment]
After digging around a bit I found out that ollama is installed in the image directly and not with the official ollama docker image:
https://github.com/open-webui/open-webui/blob/7228b39064ac28e1240bf8998f2a35535c6f7ef5/Dockerfile#L108-L130

So a workaround that worked for me is to just update ollama inside of the container directly:
```
# check the ollama version in the open-webui:ollama bundeled image
docker exec open-webui ollama --version

# update apt list
docker exec open-webui apt-get update

# update ollama to the latest version
docker exec open-webui curl -fsSL https://ollama.com/install.sh | sh

# restart container
docker restart open-webui
```

**Comment by yuchenxie4645** [comment]
I realized that I forgot to update this question. I found the solution. All you have to do is run watchtower and it will automatically update all docker images (including bundled ollama support). 

```bash
docker run --rm --volume /var/run/docker.sock:/var/run/docker.sock containrrr/watchtower --run-once open-webui
```

**Reply by RienAD** [reply]
To update your Open WebUI Docker container to the latest version, you can also follow these steps:

Stop the Running Container: First, stop the currently running Open WebUI container. You can do this by running the following command in your terminal or command prompt: 
> docker stop open-webui

Remove the Old Container: After stopping the container, you should remove it to avoid conflicts with the new version. Use the following command:

> docker rm open-webui

Pull the Latest Image: Next, pull the latest version of the Open WebUI Docker image with Ollama. You can do this with:

> docker pull ghcr.io/open-webui/open-webui:ollama

Run the Updated Container: Now, run the new container with the same configurations you used before. Here’s an example command for running it with Ollama support:

> docker run -d -p 3000:8080 -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main

Adjust the command as necessary based on your previous setup (e.g., if using specific environment variables or additional Docker options).

Verify the Update: After the container is running, check the logs to ensure it's the latest version. You can view the logs with:

> docker logs open-webui

Look for the version number in the log output to confirm the update was successful.
